test/suite/test_prepare_hs04.py

```diff
@@ -0,0 +1,217 @@
+                    self.assertRaisesException(wiredtiger.WiredTigerError, lambda:cursor.search(), preparemsg)
+            elif timestamp == 30:
+                # Keys are visible at timestamp 30.
+                self.assertEqual(cursor.search(), 0)
+
+        cursor.close()
+        self.session.commit_transaction()
+
+    def prepare_updates(self, ds):
+
+        # Commit some updates to get eviction and history store fired up
+        # Insert a ket at timestamp 1
+        commit_key = "C"
+        commit_value = b"bbbbb" * 100
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(1, self.nsessions * self.nkeys):
+            self.session.begin_transaction('isolation=snapshot')
+            key = commit_key + ds.key(self.nrows + i)
+            cursor.set_key(key)
+            cursor.set_value(commit_value)
+            self.assertEquals(cursor.insert(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(1))
+        cursor.close()
+
+        # Call checkpoint
+        self.session.checkpoint()
+
+        # Remove the committed key at timestamp 10
+        commit_value = b"bbbbb" * 100
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(1, self.nsessions * self.nkeys):
+            self.session.begin_transaction('isolation=snapshot')
+            key = commit_key + ds.key(self.nrows + i)
+            cursor.set_key(key)
+            self.assertEquals(cursor.remove(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(10))
+        cursor.close()
+
+        hs_writes_start = self.get_stat(stat.conn.cache_write_hs)
+        # Have prepared updates in multiple sessions. This should ensure writing
+        # prepared updates to the history store
+        # Insert the same key at timestamp 20, but with prepare updates.
+        sessions = [0] * self.nsessions
+        cursors = [0] * self.nsessions
+        prepare_value = b"ccccc" * 100
+        for j in range (0, self.nsessions):
+            sessions[j] = self.conn.open_session()
+            sessions[j].begin_transaction('isolation=snapshot')
+            cursors[j] = sessions[j].open_cursor(self.uri)
+            # Each session will update many consecutive keys.
+            start = (j * self.nkeys)
+            end = start + self.nkeys
+            for i in range(start, end):
+                cursors[j].set_key(commit_key + ds.key(self.nrows + i))
+                cursors[j].set_value(prepare_value)
+                self.assertEquals(cursors[j].insert(), 0)
+            sessions[j].prepare_transaction('prepare_timestamp=' + timestamp_str(20))
+
+        hs_writes = self.get_stat(stat.conn.cache_write_hs) - hs_writes_start
+
+        # Assert if not writing anything to the history store.
+        self.assertGreaterEqual(hs_writes, 0)
+
+        # Search the key with timestamp 5 and ignore_prepare=false
+        self.search_key_timestamp_and_ignore(ds, 5, "false")
```

Comment:
I think the function would be better called `search_keys_timestamp_and_ignore` since it is search many keys, not just one.  And comment `Search the key` should be updated as well.

Reply:
Fixed